File: app/src/main/java/com/app/ralaunch/controls/editor/manager/ControlEditDialogDataFiller.java
package com.app.ralaunch.controls.editor.manager;

import android.content.Context;
import android.view.View;
import android.widget.EditText;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.appcompat.widget.SwitchCompat;

import com.app.ralaunch.R;
import com.app.ralaunch.controls.ControlData;
import com.app.ralaunch.controls.KeyMapper;
import com.app.ralaunch.controls.editor.manager.ControlEditDialogVisibilityManager;
import com.google.android.material.slider.Slider;

/**
 * 控件编辑对话框数据填充管理器
 * 统一管理所有数据填充逻辑
 */
public class ControlEditDialogDataFiller {
    
    /**
     * UI元素引用接口
     */
    public interface UIReferences {
        ControlData getCurrentData();
        int getScreenWidth();
        int getScreenHeight();
        boolean isAutoSize();
        Context getContext();
    }
    
    /**
     * 填充基本信息数据
     */
    public static void fillBasicInfoData(@NonNull View view, @NonNull UIReferences refs) {
        TextView tvControlType = view.findViewById(R.id.tv_control_type);
        TextView tvControlShape = view.findViewById(R.id.tv_control_shape);
        EditText etName = view.findViewById(R.id.et_control_name);
        
        if (refs.getCurrentData() == null) return;
        
        ControlTypeManager.updateTypeDisplay(refs.getContext(), refs.getCurrentData(), tvControlType);
        ControlShapeManager.updateShapeDisplay(refs.getContext(), refs.getCurrentData(), tvControlShape, 
            view.findViewById(R.id.item_control_shape));
        
        // 摇杆模式显示（仅摇杆类型显示）
        TextView tvJoystickMode = view.findViewById(R.id.tv_joystick_mode);
        if (tvJoystickMode != null && refs.getCurrentData().type == ControlData.TYPE_JOYSTICK) {
            ControlJoystickModeManager.updateModeDisplay(refs.getContext(), refs.getCurrentData(), tvJoystickMode);
        }
        
        if (etName != null && refs.getCurrentData().name != null) {
            etName.setText(refs.getCurrentData().name);
        }
        
        // 文本内容编辑（仅文本控件显示）
        EditText etTextContent = view.findViewById(R.id.et_text_content);
        if (etTextContent != null && refs.getCurrentData().displayText != null) {
            etTextContent.setText(refs.getCurrentData().displayText);
        }
        
        // 更新基本信息选项的可见性
        ControlEditDialogVisibilityManager.updateBasicInfoOptionsVisibility(view, refs.getCurrentData());
        
        // 摇杆左右选择开关（仅摇杆类型且为SDL控制器模式或鼠标模式时显示）
        SwitchCompat switchJoystickStickSelect = view.findViewById(R.id.switch_joystick_stick_select);
        TextView tvJoystickStickSelect = view.findViewById(R.id.tv_joystick_stick_select);
        if (switchJoystickStickSelect != null && refs.getCurrentData().type == ControlData.TYPE_JOYSTICK) {
            switchJoystickStickSelect.setChecked(refs.getCurrentData().xboxUseRightStick);
            // 更新显示文本
            if (tvJoystickStickSelect != null) {
                Context context = refs.getContext();
                String text = refs.getCurrentData().xboxUseRightStick ? 
                    context.getString(R.string.editor_right_stick) : 
                    context.getString(R.string.editor_left_stick);
                tvJoystickStickSelect.setText(text);
            }
        }
        
        // 右摇杆攻击模式 RadioGroup（从全局设置读取）
        // 初始化已在 ControlEditDialogUIBinder 中处理
        
        // 触摸穿透开关
        SwitchCompat switchPassThrough = view.findViewById(R.id.switch_pass_through);
        if (switchPassThrough != null) {
            switchPassThrough.setChecked(refs.getCurrentData().passThrough);
        }
        
        // 鼠标移动范围和速度（从全局设置读取）
        // 初始化已在 ControlEditDialogUIBinder 中处理
        
    }
    
    /**
     * 填充位置大小数据
     */
    public static void fillPositionSizeData(@NonNull View view, @NonNull UIReferences refs) {
        if (refs.getCurrentData() == null) return;
        
        // 摇杆大小
        Slider sliderJoystickSize = view.findViewById(R.id.seekbar_joystick_size);
        TextView tvJoystickSizeValue = view.findViewById(R.id.tv_joystick_size_value);
        
        Slider sliderPosX = view.findViewById(R.id.seekbar_pos_x);
        TextView tvPosXValue = view.findViewById(R.id.tv_pos_x_value);
        Slider sliderPosY = view.findViewById(R.id.seekbar_pos_y);
        TextView tvPosYValue = view.findViewById(R.id.tv_pos_y_value);
        Slider sliderWidth = view.findViewById(R.id.seekbar_width);
        TextView tvWidthValue = view.findViewById(R.id.tv_width_value);
        Slider sliderHeight = view.findViewById(R.id.seekbar_height);
        TextView tvHeightValue = view.findViewById(R.id.tv_height_value);
        SwitchCompat switchAutoSize = view.findViewById(R.id.switch_auto_size);
        
        // 填充摇杆大小数据（仅摇杆类型）
        boolean isJoystick = refs.getCurrentData().type == ControlData.TYPE_JOYSTICK;
        if (sliderJoystickSize != null && isJoystick) {
            int sizePercent = (int) (refs.getCurrentData().width / refs.getScreenWidth() * 100);
            sizePercent = Math.max(1, Math.min(100, sizePercent));
            sliderJoystickSize.setValue(sizePercent);
            if (tvJoystickSizeValue != null) tvJoystickSizeValue.setText(sizePercent + "%");
        }
        
        if (sliderPosX != null) {
            int xPercent = (int) (refs.getCurrentData().x / refs.getScreenWidth() * 100);
            xPercent = Math.max(0, Math.min(100, xPercent)); // 确保在有效范围内
            sliderPosX.setValue(xPercent);
            if (tvPosXValue != null) tvPosXValue.setText(xPercent + "%");
        }
        
        if (sliderPosY != null) {
            int yPercent = (int) (refs.getCurrentData().y / refs.getScreenHeight() * 100);
            yPercent = Math.max(0, Math.min(100, yPercent)); // 确保在有效范围内
            sliderPosY.setValue(yPercent);
            if (tvPosYValue != null) tvPosYValue.setText(yPercent + "%");
        }
        
        if (sliderWidth != null) {
            int widthPercent = (int) (refs.getCurrentData().width / refs.getScreenWidth() * 100);
            widthPercent = Math.max(1, Math.min(100, widthPercent)); // 确保在有效范围内，最小1%
            sliderWidth.setValue(widthPercent);
            if (tvWidthValue != null) tvWidthValue.setText(widthPercent + "%");
        }
        
        if (sliderHeight != null) {
            int heightPercent = (int) (refs.getCurrentData().height / refs.getScreenHeight() * 100);
            heightPercent = Math.max(1, Math.min(100, heightPercent)); // 确保在有效范围内，最小1%
            sliderHeight.setValue(heightPercent);
            if (tvHeightValue != null) tvHeightValue.setText(heightPercent + "%");
        }
        
        if (switchAutoSize != null) {
            // 检查当前宽高是否相等，如果相等则默认启用自适应
            boolean isAutoSize = Math.abs(refs.getCurrentData().width - refs.getCurrentData().height) < 1.0f;
            switchAutoSize.setChecked(isAutoSize);
        }
    }
    
    /**
     * 填充外观样式数据
     */
    public static void fillAppearanceData(@NonNull View view, @NonNull UIReferences refs) {
        if (refs.getCurrentData() == null) return;
        
        ControlData data = refs.getCurrentData();
        
        // 根据控件类型更新透明度标题和描述
        TextView tvOpacityTitle = view.findViewById(R.id.tv_opacity_title);
        TextView tvOpacityDesc = view.findViewById(R.id.tv_opacity_desc);
        Context context = refs.getContext();
        if (data.type == ControlData.TYPE_JOYSTICK) {
            if (tvOpacityTitle != null) {
                tvOpacityTitle.setText(context.getString(R.string.editor_joystick_bg_opacity));
            }
            if (tvOpacityDesc != null) {
                tvOpacityDesc.setText(context.getString(R.string.editor_joystick_bg_opacity_desc));
            }
        } else {
            if (tvOpacityTitle != null) {
                tvOpacityTitle.setText(context.getString(R.string.editor_opacity));
            }
            if (tvOpacityDesc != null) {
                tvOpacityDesc.setText(context.getString(R.string.editor_opacity_desc));
            }
        }
        
        // 透明度设置（使用统一管理器）
        ControlEditDialogSeekBarManager.fillSeekBarSetting(view,
            R.id.seekbar_opacity,
            R.id.tv_opacity_value,
            ControlEditDialogSeekBarManager.createPercentConfig(
                data,
                new ControlEditDialogSeekBarManager.ValueSetter() {
                    @Override
                    public float get(ControlData d) {
                        return d.opacity;
                    }
                    
                    @Override
                    public void set(ControlData d, float value) {
                        d.opacity = value;
                    }
                },
                () -> {} // 填充时不需要通知更新
            )
        );
        
        // 边框透明度设置（使用统一管理器，完全独立）
        ControlEditDialogSeekBarManager.fillSeekBarSetting(view,
            R.id.seekbar_border_opacity,
            R.id.tv_border_opacity_value,
            ControlEditDialogSeekBarManager.createPercentConfig(
                data,
                new ControlEditDialogSeekBarManager.ValueSetter() {
                    @Override
                    public float get(ControlData d) {
                        // 边框透明度完全独立，默认1.0（完全不透明）
                        return d.borderOpacity != 0 ? d.borderOpacity : 1.0f;
                    }
                    
                    @Override
                    public void set(ControlData d, float value) {
                        d.borderOpacity = value;
                    }
                },
                () -> {} // 填充时不需要通知更新
            )
        );
        
        // 文本透明度设置（使用统一管理器，仅按钮和文本控件显示）
        View cardTextOpacity = view.findViewById(R.id.card_text_opacity);
        if (cardTextOpacity != null && cardTextOpacity.getVisibility() == View.VISIBLE) {
            ControlEditDialogSeekBarManager.fillSeekBarSetting(view,
                R.id.seekbar_text_opacity,
                R.id.tv_text_opacity_value,
                ControlEditDialogSeekBarManager.createPercentConfig(
                    data,
                    new ControlEditDialogSeekBarManager.ValueSetter() {
                        @Override
                        public float get(ControlData d) {
                            // 文本透明度完全独立，默认1.0（完全不透明）
                            return d.textOpacity != 0 ? d.textOpacity : 1.0f;
                        }
                        
                        @Override
                        public void set(ControlData d, float value) {
                            d.textOpacity = value;
                        }
                    },
                    () -> {} // 填充时不需要通知更新
                )
            );
        }
        
        SwitchCompat switchVisible = view.findViewById(R.id.switch_visible);
        if (switchVisible != null) {
            switchVisible.setChecked(data.visible);
        }
        
        // 更新外观选项的可见性（根据控件类型和形状）
        ControlEditDialogVisibilityManager.updateAppearanceOptionsVisibility(view, data);
        
        // 圆角半径设置（仅在矩形形状时显示）
        View cardCornerRadius = view.findViewById(R.id.card_corner_radius);
        if (cardCornerRadius != null && cardCornerRadius.getVisibility() == View.VISIBLE) {
            Slider sliderCornerRadius = view.findViewById(R.id.seekbar_corner_radius);
            TextView tvCornerRadiusValue = view.findViewById(R.id.tv_corner_radius_value);
            if (sliderCornerRadius != null && tvCornerRadiusValue != null) {
                int cornerRadius = (int) data.cornerRadius;
                // 确保在 Slider 的有效范围内
                cornerRadius = (int) Math.max(sliderCornerRadius.getValueFrom(), 
                    Math.min(sliderCornerRadius.getValueTo(), cornerRadius));
                sliderCornerRadius.setValue(cornerRadius);
                tvCornerRadiusValue.setText(cornerRadius + "dp");
            }
        }
        
        // 摇杆中心透明度（使用统一管理器，仅在摇杆类型时显示）
        View cardStickOpacity = view.findViewById(R.id.card_stick_opacity);
        if (cardStickOpacity != null && cardStickOpacity.getVisibility() == View.VISIBLE) {
            ControlEditDialogSeekBarManager.fillSeekBarSetting(view,
                R.id.seekbar_stick_opacity,
                R.id.tv_stick_opacity_value,
                ControlEditDialogSeekBarManager.createPercentConfig(
                    data,
                    new ControlEditDialogSeekBarManager.ValueSetter() {
                        @Override
                        public float get(ControlData d) {
                            return d.stickOpacity != 0 ? d.stickOpacity : 1.0f;
                        }
                        
                        @Override
                        public void set(ControlData d, float value) {
                            d.stickOpacity = value;
                        }
                    },
                    () -> {} // 填充时不需要通知更新
                )
            );
        }
        
        // 摇杆圆心大小（使用统一管理器，仅在摇杆类型时显示）
        View cardStickKnobSize = view.findViewById(R.id.card_stick_knob_size);
        if (cardStickKnobSize != null && cardStickKnobSize.getVisibility() == View.VISIBLE) {
            ControlEditDialogSeekBarManager.fillSeekBarSetting(view,
                R.id.seekbar_stick_knob_size,
                R.id.tv_stick_knob_size_value,
                ControlEditDialogSeekBarManager.createPercentConfig(
                    data,
                    new ControlEditDialogSeekBarManager.ValueSetter() {
                        @Override
                        public float get(ControlData d) {
                            return d.stickKnobSize != 0 ? d.stickKnobSize : 0.4f;
                        }
                        
                        @Override
                        public void set(ControlData d, float value) {
                            d.stickKnobSize = value;
                        }
                    },
                    () -> {} // 填充时不需要通知更新
                )
            );
        }
        
        
        // 更新颜色视图
        View viewBgColor = view.findViewById(R.id.view_bg_color);
        View viewStrokeColor = view.findViewById(R.id.view_stroke_color);
        if (viewBgColor != null) {
            ControlColorManager.updateColorView(viewBgColor, data.bgColor, 8, 2);
        }
        if (viewStrokeColor != null) {
            ControlColorManager.updateColorView(viewStrokeColor, data.strokeColor, 8, 2);
        }
        
    }
    
    /**
     * 填充键值设置数据
     */
    public static void fillKeymapData(@NonNull View view, @NonNull UIReferences refs) {
        if (refs.getCurrentData() == null) return;
        
        ControlData data = refs.getCurrentData();
        
        // 更新键值设置选项的可见性（根据控件类型）
        ControlEditDialogVisibilityManager.updateKeymapOptionsVisibility(view, data);
        
        // 填充按钮键值数据
        fillButtonKeymap(view, data);
    }
    
    /**
     * 填充普通按钮键值数据
     */
    private static void fillButtonKeymap(@NonNull View view, @NonNull ControlData data) {
        TextView tvKeyName = view.findViewById(R.id.tv_key_name);
        if (tvKeyName != null) {
            String keyName = KeyMapper.getKeyName(data.keycode);
            tvKeyName.setText(keyName);
        }
        
        SwitchCompat switchToggleMode = view.findViewById(R.id.switch_toggle_mode);
        if (switchToggleMode != null) {
            switchToggleMode.setChecked(data.isToggle);
        }
    }
    
}

File: app/src/main/java/com/app/ralaunch/controls/editors/managers/ControlEditDialogDataFiller.kt
package com.app.ralaunch.controls.editors.managers

import android.content.Context
import android.view.View
import android.widget.EditText
import android.widget.TextView
import androidx.appcompat.widget.SwitchCompat
import com.app.ralaunch.R
import com.app.ralaunch.controls.KeyMapper
import com.app.ralaunch.controls.data.ControlData
import com.google.android.material.slider.Slider
import kotlin.math.max
import kotlin.math.min

/**
 * 控件编辑对话框数据填充管理器
 * 统一管理所有数据填充逻辑
 */
object ControlEditDialogDataFiller {
    /**
     * 填充基本信息数据
     */
    @JvmStatic
    fun fillBasicInfoData(view: View, refs: UIReferences) {
        val tvControlType = view.findViewById<TextView?>(R.id.tv_control_type)
        val tvControlShape = view.findViewById<TextView?>(R.id.tv_control_shape)
        val etName = view.findViewById<EditText?>(R.id.et_control_name)

        val data = refs.currentData ?: return

        ControlTypeManager.updateTypeDisplay(refs.context, data, tvControlType)
        ControlShapeManager.updateShapeDisplay(
            refs.context, data, tvControlShape,
            view.findViewById(R.id.item_control_shape)
        )

        // 摇杆模式显示（仅摇杆类型显示）
        val tvJoystickMode = view.findViewById<TextView?>(R.id.tv_joystick_mode)
        if (tvJoystickMode != null && data is ControlData.Joystick) {
            ControlJoystickModeManager.updateModeDisplay(refs.context, data, tvJoystickMode)
        }

        etName?.setText(data.name)

        // 文本内容编辑（仅文本控件显示）
        val etTextContent = view.findViewById<EditText?>(R.id.et_text_content)
        if (etTextContent != null && data is ControlData.Text) {
            etTextContent.setText(data.displayText)
        }

        // 更新基本信息选项的可见性
        ControlEditDialogVisibilityManager.updateAllOptionsVisibility(view, data)

        // 摇杆左右选择开关（仅摇杆类型且为SDL控制器模式或鼠标模式时显示）
        val switchJoystickStickSelect = view.findViewById<SwitchCompat?>(R.id.switch_joystick_stick_select)
        val tvJoystickStickSelect = view.findViewById<TextView?>(R.id.tv_joystick_stick_select)
        if (switchJoystickStickSelect != null && data is ControlData.Joystick) {
            switchJoystickStickSelect.isChecked = data.isRightStick
            // 更新显示文本
            tvJoystickStickSelect?.let {
                val text = if (data.isRightStick) {
                    refs.context.getString(R.string.editor_right_stick)
                } else {
                    refs.context.getString(R.string.editor_left_stick)
                }
                it.text = text
            }
        }

        // 触摸穿透开关
        val switchPassThrough = view.findViewById<SwitchCompat?>(R.id.switch_pass_through)
        switchPassThrough?.isChecked = data.isPassThrough

        // 双击模拟摇杆开关（仅触摸板显示）
        val switchDoubleClickJoystick = view.findViewById<SwitchCompat?>(R.id.switch_double_click_joystick)
        if (switchDoubleClickJoystick != null && data is ControlData.TouchPad) {
            switchDoubleClickJoystick.isChecked = data.isDoubleClickSimulateJoystick
        }

        // 鼠标滚轮设置（绑定UI，无论什么数据类型）
        // 滚轮方向开关
        val switchMouseWheelOrientation = view.findViewById<SwitchCompat?>(R.id.switch_mousewheel_orientation)
        if (data is ControlData.MouseWheel) {
            switchMouseWheelOrientation?.isChecked = (data.orientation == ControlData.MouseWheel.Orientation.HORIZONTAL)
        } else {
            switchMouseWheelOrientation?.isChecked = false
        }

        // 反转方向开关
        val switchMouseWheelReverse = view.findViewById<SwitchCompat?>(R.id.switch_mousewheel_reverse)
        if (data is ControlData.MouseWheel) {
            switchMouseWheelReverse?.isChecked = data.reverseDirection
        } else {
            switchMouseWheelReverse?.isChecked = false
        }

        // 滚轮灵敏度
        val sliderMouseWheelSensitivity = view.findViewById<Slider?>(R.id.slider_mousewheel_sensitivity)
        val tvMouseWheelSensitivityValue = view.findViewById<TextView?>(R.id.tv_mousewheel_sensitivity_value)
        if (sliderMouseWheelSensitivity != null) {
            if (data is ControlData.MouseWheel) {
                sliderMouseWheelSensitivity.value = data.scrollSensitivity
                tvMouseWheelSensitivityValue?.text = String.format("%.1f", data.scrollSensitivity)
            } else {
                sliderMouseWheelSensitivity.value = 15.0f
                tvMouseWheelSensitivityValue?.text = "15.0"
            }
        }

        // 滚轮速度倍率
        val sliderMouseWheelRatio = view.findViewById<Slider?>(R.id.slider_mousewheel_ratio)
        val tvMouseWheelRatioValue = view.findViewById<TextView?>(R.id.tv_mousewheel_ratio_value)
        if (sliderMouseWheelRatio != null) {
            if (data is ControlData.MouseWheel) {
                sliderMouseWheelRatio.value = data.scrollRatio
                tvMouseWheelRatioValue?.text = "${(data.scrollRatio * 100).toInt()}%"
            } else {
                sliderMouseWheelRatio.value = 1.0f
                tvMouseWheelRatioValue?.text = "100%"
            }
        }
    }

    /**
     * 填充位置大小数据
     */
    @JvmStatic
    fun fillPositionSizeData(view: View, refs: UIReferences) {
        val data = refs.currentData ?: return

        // 摇杆大小
        val sliderJoystickSize = view.findViewById<Slider?>(R.id.seekbar_joystick_size)
        val tvJoystickSizeValue = view.findViewById<TextView?>(R.id.tv_joystick_size_value)

        val sliderPosX = view.findViewById<Slider?>(R.id.seekbar_pos_x)
        val tvPosXValue = view.findViewById<TextView?>(R.id.tv_pos_x_value)
        val sliderPosY = view.findViewById<Slider?>(R.id.seekbar_pos_y)
        val tvPosYValue = view.findViewById<TextView?>(R.id.tv_pos_y_value)
        val sliderWidth = view.findViewById<Slider?>(R.id.seekbar_width)
        val tvWidthValue = view.findViewById<TextView?>(R.id.tv_width_value)
        val sliderHeight = view.findViewById<Slider?>(R.id.seekbar_height)
        val tvHeightValue = view.findViewById<TextView?>(R.id.tv_height_value)
        val switchAutoSize = view.findViewById<SwitchCompat?>(R.id.switch_auto_size)

        // 填充摇杆大小数据（仅摇杆类型）
        if (sliderJoystickSize != null && data is ControlData.Joystick) {
            val sizePercent = max(1, min(100, (data.width * 100).toInt()))
            sliderJoystickSize.value = sizePercent.toFloat()
            tvJoystickSizeValue?.text = "$sizePercent%"
        }

        sliderPosX?.let {
            val xPercent = max(0, min(100, (data.x * 100).toInt()))
            it.value = xPercent.toFloat()
            tvPosXValue?.text = "$xPercent%"
        }

        sliderPosY?.let {
            val yPercent = max(0, min(100, (data.y * 100).toInt()))
            it.value = yPercent.toFloat()
            tvPosYValue?.text = "$yPercent%"
        }

        sliderWidth?.let {
            val widthPercent = max(1, min(100, (data.width * 100).toInt()))
            it.value = widthPercent.toFloat()
            tvWidthValue?.text = "$widthPercent%"
        }

        sliderHeight?.let {
            val heightPercent = max(1, min(100, (data.height * 100).toInt()))
            it.value = heightPercent.toFloat()
            tvHeightValue?.text = "$heightPercent%"
        }

        switchAutoSize?.let {
            it.isChecked = data.isSizeRatioLocked
        }
    }

    /**
     * 填充外观样式数据
     */
    @JvmStatic
    fun fillAppearanceData(view: View, refs: UIReferences) {
        val data = refs.currentData ?: return

        // 根据控件类型更新透明度标题和描述
        val tvOpacityTitle = view.findViewById<TextView?>(R.id.tv_opacity_title)
        val tvOpacityDesc = view.findViewById<TextView?>(R.id.tv_opacity_desc)
        val context = refs.context
        if (data is ControlData.Joystick) {
            tvOpacityTitle?.setText(context.getString(R.string.editor_joystick_bg_opacity))
            tvOpacityDesc?.setText(context.getString(R.string.editor_joystick_bg_opacity_desc))
        } else {
            tvOpacityTitle?.setText(context.getString(R.string.editor_opacity))
            tvOpacityDesc?.setText(context.getString(R.string.editor_opacity_desc))
        }

        // 透明度设置（使用统一管理器）
        ControlEditDialogSeekBarManager.fillSeekBarSetting(
            view,
            R.id.seekbar_opacity,
            R.id.tv_opacity_value,
            ControlEditDialogSeekBarManager.createPercentConfig(
                data,
                object : ControlEditDialogSeekBarManager.ValueSetter {
                    override fun get(data: ControlData?): Float = data?.opacity ?: 0.5f
                    override fun set(data: ControlData?, value: Float) {
                        data?.let { it.opacity = value }
                    }
                },
                object : ControlEditDialogSeekBarManager.UpdateNotifier {
                    override fun notifyUpdate() {} // 填充时不需要通知更新
                }
            )
        )

        // 边框透明度设置（使用统一管理器，完全独立）
        ControlEditDialogSeekBarManager.fillSeekBarSetting(
            view,
            R.id.seekbar_border_opacity,
            R.id.tv_border_opacity_value,
            ControlEditDialogSeekBarManager.createPercentConfig(
                data,
                object : ControlEditDialogSeekBarManager.ValueSetter {
                    override fun get(data: ControlData?): Float {
                        // 边框透明度完全独立，默认1.0（完全不透明），0是有效值
                        return data?.borderOpacity ?: 1.0f
                    }

                    override fun set(data: ControlData?, value: Float) {
                        data?.let { it.borderOpacity = value }
                    }
                },
                object : ControlEditDialogSeekBarManager.UpdateNotifier {
                    override fun notifyUpdate() {} // 填充时不需要通知更新
                }
            )
        )

        // 文本透明度设置（使用统一管理器，仅按钮和文本控件显示）
        val cardTextOpacity = view.findViewById<View?>(R.id.card_text_opacity)
        if (cardTextOpacity != null && cardTextOpacity.visibility == View.VISIBLE) {
            ControlEditDialogSeekBarManager.fillSeekBarSetting(
                view,
                R.id.seekbar_text_opacity,
                R.id.tv_text_opacity_value,
                ControlEditDialogSeekBarManager.createPercentConfig(
                    data,
                    object : ControlEditDialogSeekBarManager.ValueSetter {
                        override fun get(data: ControlData?): Float {
                            // 文本透明度完全独立，默认1.0（完全不透明），0是有效值
                            return data?.textOpacity ?: 1.0f
                        }

                        override fun set(data: ControlData?, value: Float) {
                            data?.let { it.textOpacity = value }
                        }
                    },
                    object : ControlEditDialogSeekBarManager.UpdateNotifier {
                        override fun notifyUpdate() {} // 填充时不需要通知更新
                    }
                )
            )
        }

        val switchVisible = view.findViewById<SwitchCompat?>(R.id.switch_visible)
        switchVisible?.isChecked = data.isVisible

        // 更新外观选项的可见性（根据控件类型和形状）
        ControlEditDialogVisibilityManager.updateAllOptionsVisibility(view, data)

        // 圆角半径设置（仅在矩形形状时显示）
        val cardCornerRadius = view.findViewById<View?>(R.id.card_corner_radius)
        if (cardCornerRadius != null && cardCornerRadius.visibility == View.VISIBLE) {
            val sliderCornerRadius = view.findViewById<Slider?>(R.id.seekbar_corner_radius)
            val tvCornerRadiusValue = view.findViewById<TextView?>(R.id.tv_corner_radius_value)
            if (sliderCornerRadius != null && tvCornerRadiusValue != null) {
                val cornerRadius = max(
                    sliderCornerRadius.valueFrom,
                    min(sliderCornerRadius.valueTo, data.cornerRadius)
                ).toInt()
                sliderCornerRadius.value = cornerRadius.toFloat()
                tvCornerRadiusValue.text = "${cornerRadius}dp"
            }
        }

        // 摇杆中心透明度（使用统一管理器，仅在摇杆类型时显示）
        val cardStickOpacity = view.findViewById<View?>(R.id.card_stick_opacity)
        if (cardStickOpacity != null && cardStickOpacity.visibility == View.VISIBLE && data is ControlData.Joystick) {
            ControlEditDialogSeekBarManager.fillSeekBarSetting(
                view,
                R.id.seekbar_stick_opacity,
                R.id.tv_stick_opacity_value,
                ControlEditDialogSeekBarManager.createPercentConfig(
                    data,
                    object : ControlEditDialogSeekBarManager.ValueSetter {
                        override fun get(data: ControlData?): Float {
                            // 摇杆圆心透明度，默认1.0（完全不透明），0是有效值
                            return (data as? ControlData.Joystick)?.stickOpacity ?: 1.0f
                        }

                        override fun set(data: ControlData?, value: Float) {
                            if (data is ControlData.Joystick) {
                                data.stickOpacity = value
                            }
                        }
                    },
                    object : ControlEditDialogSeekBarManager.UpdateNotifier {
                        override fun notifyUpdate() {} // 填充时不需要通知更新
                    }
                )
            )
        }

        // 摇杆圆心大小（使用统一管理器，仅在摇杆类型时显示）
        val cardStickKnobSize = view.findViewById<View?>(R.id.card_stick_knob_size)
        if (cardStickKnobSize != null && cardStickKnobSize.visibility == View.VISIBLE && data is ControlData.Joystick) {
            ControlEditDialogSeekBarManager.fillSeekBarSetting(
                view,
                R.id.seekbar_stick_knob_size,
                R.id.tv_stick_knob_size_value,
                ControlEditDialogSeekBarManager.createPercentConfig(
                    data,
                    object : ControlEditDialogSeekBarManager.ValueSetter {
                        override fun get(data: ControlData?): Float {
                            // 摇杆圆心大小比例，默认0.4，0是有效值（可以让摇杆圆心不可见）
                            return (data as? ControlData.Joystick)?.stickKnobSize ?: 0.4f
                        }

                        override fun set(data: ControlData?, value: Float) {
                            if (data is ControlData.Joystick) {
                                data.stickKnobSize = value
                            }
                        }
                    },
                    object : ControlEditDialogSeekBarManager.UpdateNotifier {
                        override fun notifyUpdate() {} // 填充时不需要通知更新
                    }
                )
            )
        }

        // 更新颜色视图
        val viewBgColor = view.findViewById<View?>(R.id.view_bg_color)
        val viewStrokeColor = view.findViewById<View?>(R.id.view_stroke_color)
        viewBgColor?.let {
            ControlColorManager.updateColorView(it, data.bgColor, 8f, 2f)
        }
        viewStrokeColor?.let {
            ControlColorManager.updateColorView(it, data.strokeColor, 8f, 2f)
        }
    }

    /**
     * 填充键值设置数据
     */
    @JvmStatic
    fun fillKeymapData(view: View, refs: UIReferences) {
        val data = refs.currentData ?: return

        // 更新键值设置选项的可见性（根据控件类型）
        ControlEditDialogVisibilityManager.updateAllOptionsVisibility(view, data)

        // 填充按钮键值数据
        fillButtonKeymap(view, data)
    }

    /**
     * 填充普通按钮键值数据
     */
    private fun fillButtonKeymap(view: View, data: ControlData) {
        val tvKeyName = view.findViewById<TextView?>(R.id.tv_key_name)
        if (tvKeyName != null && data is ControlData.Button) {
            val keyName = KeyMapper.getKeyName(data.keycode)
            tvKeyName.text = keyName
        }

        val switchToggleMode = view.findViewById<SwitchCompat?>(R.id.switch_toggle_mode)
        if (switchToggleMode != null && data is ControlData.Button) {
            switchToggleMode.isChecked = data.isToggle
        }
    }

    /**
     * UI元素引用接口
     */
    interface UIReferences {
        val currentData: ControlData?
        val screenWidth: Int
        val screenHeight: Int
        val context: Context
    }
}

